x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/node/label_node/label_node.tsx

```diff
@@ -0,0 +1,164 @@
+import { LabelNodeDetails } from './label_node_details';
+
+export const TEST_SUBJ_CONTAINER = 'label-node-container';
+export const TEST_SUBJ_SHAPE = 'label-node-shape';
+export const TEST_SUBJ_TOOLTIP = 'label-node-tooltip';
+export const TEST_SUBJ_HANDLE = 'label-node-handle';
+export const TEST_SUBJ_EXPAND_BTN = 'label-node-expand-btn';
+export const TEST_SUBJ_HOVER_OUTLINE = 'label-node-hover-outline';
+
+export const LabelNode = memo<NodeProps>((props: NodeProps) => {
+  const {
+    id,
+    color,
+    eventsCount,
+    alertsCount,
+    label,
+    interactive,
+    ips,
+    countryCodes,
+    nodeClick,
+    expandButtonClick,
+  } = props.data as LabelNodeViewModel;
+
+  const { euiTheme } = useEuiTheme();
+  const shadow = useEuiShadow('m', { property: 'filter' });
+
+  const text = label ? label : id;
+
+  const { backgroundColor, borderColor, textColor } = useMemo(
+    () => getLabelColors(color, euiTheme),
+    [color, euiTheme]
+  );
+
+  const numEvents = eventsCount ?? 0;
+  const numAlerts = alertsCount ?? 0;
+  // TODO Remove color when API returns `eventsCount` and `alertsCount`
+  const analysis = analyzeDocuments({ color, eventsCount: numEvents, alertsCount: numAlerts });
+  const shouldShowTooltip = numEvents > BADGES_LIMIT || numAlerts > BADGES_LIMIT;
+
+  return (
+    <>
+      <EuiToolTip
+        display="block"
+        title={shouldShowTooltip ? text : null}
+        content={shouldShowTooltip ? <LabelNodeTooltipContent analysis={analysis} /> : null}
+        position="top"
+        data-test-subj={TEST_SUBJ_TOOLTIP}
+      >
+        <LabelNodeContainer data-test-subj={TEST_SUBJ_CONTAINER}>
+          {interactive && (
+            <LabelShapeOnHover
+              data-test-subj={TEST_SUBJ_HOVER_OUTLINE}
+              color={
+                analysis.isSingleAlert || analysis.isGroupOfAlerts
+                  ? euiTheme.colors.danger
+                  : euiTheme.colors.primary
+              }
+            />
+          )}
+          <LabelShape
+            data-test-subj={TEST_SUBJ_SHAPE}
+            backgroundColor={backgroundColor}
+            borderColor={borderColor}
+            textAlign="center"
+            shadow={shadow}
+          >
+            <div
+              css={css`
+                display: flex;
+                align-items: center;
+                justify-content: space-between;
+                width: 100%;
+                gap: ${euiTheme.size.xs};
+              `}
+            >
+              <EuiText
+                color={textColor}
+                css={css`
+                  flex: 1;
+                  min-width: 0;
+                  text-overflow: ellipsis;
+                  font-weight: ${euiTheme.font.weight.semiBold};
+                  font-size: ${euiTheme.font.scale.xs * 10.5}px;
+                `}
+              >
+                <EuiTextTruncate truncation="end" text={text}>
+                  {(truncatedText) => truncatedText}
```

Comment:
I'm not sure this line is necessary
When using EuiTextTruncate you can pass it text prop and that's it